sql/cagg_migrate.sql

```diff
@@ -596,3 +596,8 @@ BEGIN
     WHERE mat_hypertable_id OPERATOR(pg_catalog.=) _cagg_data.mat_hypertable_id;
 END;
 $BODY$;
+
+-- Migrate a CAgg which is using the experimental time_bucket_ng function
+-- into a CAgg using the regular time_bucket function
+CREATE OR REPLACE PROCEDURE @extschema@.cagg_migrate_to_time_bucket(cagg REGCLASS)
+   AS '@MODULE_PATHNAME@', 'ts_continuous_agg_migrate_to_time_bucket' LANGUAGE C;
```

Comment:
Wondering if we should expose it to the users and let them migrate by themselves or do it only once during the extension UPDATE.

Reply:
At least, in the first release, I would just expose the function and mention it in the changelog. However, before we remove the `time_bucket_ng` code, we should (a) block the extension upgrade or (b) run the migration automatically. 

I would love to get some more feedback about real-world CAggs and migrations before we run it on all existing CAggs (it might block the extension upgrade if the function errors for some unexpected reason).

Reply:
Maybe querying `schemata` can give us the answer for this question.

Reply:
Auto migration script is added